app/Console/Commands/YouTubeVideosAndPlaylists.php

```diff
@@ -36,16 +36,26 @@ private function importUploads(): Collection
         $progress = !$this->output->isDebug() ? $this->output->createProgressBar($this->youtube->uploadCount()) : null;
         $progress?->start();
         $sourceIds = collect();
-        foreach ($this->youtube->uploads(fields: 'items/snippet(resourceId/videoId,title)') as $upload) {
+        $fields = 'items(status,snippet(resourceId/videoId,title,description))';
+        foreach ($this->youtube->uploads($fields) as $upload) {
             $videoId = $upload['snippet']['resourceId']['videoId'];
-            $sourceIds->push($videoId);
-            Video::updateOrCreate(
+            Video::withTrashed()->updateOrCreate(
                 ['youtube_id' => $videoId],
                 [
                     'youtube_id' => $videoId,
                     'title' => $upload['snippet']['title'],
+                    'description' => $upload['snippet']['description'],
                 ],
             );
+            // Mark deleted videos as such in the db.
+            if (
+                $upload['snippet']['title'] == 'Deleted video'
+                && $upload['snippet']['description'] == 'This video is unavailable.'
```

Comment:
It's kinda weird that the way they mark a video as deleted is by setting its title to 'Deleted video' and its description to 'This video is unavailable.' instead of, like, a boolean `is_deleted` or a `deleted_at` date 🤷